lib/rex/ui/text/dispatcher_shell.rb

```diff
@@ -422,12 +422,7 @@ def tab_complete_stub(str, quote: nil)
       end
     }
 
-    # Verify that our search string is a valid regex
-    begin
-      Regexp.compile(str,Regexp::IGNORECASE)
-    rescue RegexpError
-      str = Regexp.escape(str)
-    end
+    str = Regexp.escape(str)
 
     # @todo - This still doesn't fix some Regexp warnings:
     # ./lib/rex/ui/text/dispatcher_shell.rb:171: warning: regexp has `]' without escape
```

Comment:
I just realised what this comment now means:
```ruby
2.7.2 :020 > Regexp.compile('testing]', Regexp::IGNORECASE)
(irb):20: warning: regular expression has ']' without escape: /testing]/
 => /testing]/i 
```

It's roughly the same warning that you're getting with the `\P` in `C:\Pro`:
```ruby
2.7.2 :007 > Regexp.compile('C:\Pro', Regexp::IGNORECASE)
(irb):7: warning: invalid Unicode Property \P: /C:\Pro/
 => /C:\Pro/i 
2.7.2 :008 > Regexp.compile('\P', Regexp::IGNORECASE)
(irb):8: warning: invalid Unicode Property \P: /\P/
 => /\P/i 
```

Reply:
As discussed, let's delete this functionality and keep to just the `word.downcase.start_with?` functionality

```diff
-      word.downcase.start_with?(str.downcase) || word =~ /^#{str}/i
+      word.downcase.start_with?(str.downcase)
```

Reply:
Got those changes added